src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDecreaseInheritedMemberVisibilityTests.Fixer.cs

```diff
@@ -12,22 +11,387 @@ public class DoNotDecreaseInheritedMemberVisibilityFixerTests : CodeFixTestBase
         {
-            return new CSharpDoNotDecreaseInheritedMemberVisibilityFixer();
+            return new DoNotDecreaseInheritedMemberVisibilityFixer();
+        }
+
+        [Fact]
+        public void DecreaseCSharpMemberVisibility()
+        {
+            VerifyCSharpFix(
+// Original Code                
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    private new int MyProp3;
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3;
+}");
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    internal new int MyProp3 { get; set; }
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { get; set; }
+}"
+);
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { get; private set; }
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { get; set; }
+}");
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public void MyProp3() {}
+}
+
+public class DerivedClass : BaseClass
+{
+    private void MyProp3() {}
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public void MyProp3() {}
+}
+
+public class DerivedClass : BaseClass
+{
+    public void MyProp3() {}
+}");
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    protected void MyProp3() {}
+}
+
+public class DerivedClass : BaseClass
+{
+    private void MyProp3() {}
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    protected void MyProp3() {}
+}
+
+public class DerivedClass : BaseClass
+{
+    protected void MyProp3() {}
+}");
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public int MyProp3 { protected get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { private get; private set; }
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3 { protected get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { protected get; set; }
+}");
+
+            VerifyCSharpFix(
+// Original Code
+@"public class BaseClass
+{
+    public int MyProp3 { protected get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    internal new int MyProp3 { get; private set; }
+}",
+
+// Fixed Code
+@"public class BaseClass
+{
+    public int MyProp3 { protected get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 { get; set; }
+}");
+
```

Comment:
Good point. The old rule actually only fires for methods that hide base implementations (since changing visibility with an override is already a compiler error) but I should add negative tests for this scenario.